### PR TITLE
*: add TiKV overview, installation, and client driver

### PR DESCRIPTION
**Pull request by lilin90**
@siddontang @c4pt0r @disksing @queenypingcap PTAL

### CODE REVIEW

**Comment by c4pt0r** [comment]
> Currently, TiKV provides two APIs: the Raw Key-Value API and the Transactional Key-Value API. This guide uses the APIs written in Go. To write the API, you need to install Go:
> 
>     Make sure ​Go​ (version 1.8+) is installed.
>     Ensure your $GOPATH is set. (For example, export GOPATH=$HOME/go)

No need to tell users how to install Go or even how to use API.
I think this document should focus on deploy tikv, move all the demo code to another document, keep this doc short.

**Comment by c4pt0r** [comment]
Quick Guide should be short and clear. Nobody actually reads the quick guide, they just want to copy the commands, and make it run as quick as possible.

File: tikv/tikv-quick-start-guide.md [hunk 1]
```diff
@@ -0,0 +1,512 @@
...
+
+```
+./pd-ctl store -d -u http://192.168.199.113:2379
+```
+
+The result displays the store count and detailed information regarding each store.
+
+## Try TiKV using the Raw Key-Value API and Transactional Key-Value API
+
+This section introduces two types of APIs and guides you through how to use the two APIs in TiKV using two examples. The usage examples are based on the [deployment of TiKV on multiple nodes](#deploy-the-tikv-cluster-on-multiple-nodes-for-test) in the above section. You can also quickly try the two APIs on a single machine.
```
Comment (zhangjinpeng1987) [inline tikv/tikv-quick-start-guide.md]: two types of APIs?

**Comment by queenypingcap** [comment]
LGTM

File: tikv/get-started/go-client-api.md [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+
+    Here is an example to increase a key-value by 1:
+
+    ```bash
+    func increase(storage kv.Storage, key []byte) error {
+        txn, err := storage.Begin()
+        if err != nil {
+            return err
+        }
+    defer txn.Rollback()
```
Comment (siddontang) [inline tikv/get-started/go-client-api.md]: alignment

File: tikv/get-started/go-client-api.md [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+    "fmt"
+    "strconv"
+
+    "github.com/pingcap/tidb/kv"
+    "github.com/pingcap/tidb/store/tikv"
+)
+
+// if key not found, set value to zero
+// else increase the value
+func increase(storage kv.Storage, key []byte) error {
```
Comment (siddontang) [inline tikv/get-started/go-client-api.md]: duplicated with above example, I guess you can remove the above one.

File: tikv/get-started/go-client-api.md [hunk 4]
```diff
@@ -0,0 +1,328 @@
...
+
+## Try the Raw Key-Value API
+
+To use the Raw Key-Value API in applications developed by golang, take the following steps:
+
+1. Install the necessary packages.
+
+    ```bash
+    go get -v -u github.com/pingcap/tidb/config
+    go get -v -u github.com/pingcap/tidb/store/tikv
```
[the remark this answers is not in the document]
Comment (lilin90) [reply tikv/get-started/go-client-api.md]: Yeah, I've tried it. OR what else do you want to add?

**Comment by lilin90** [comment]
@siddontang @queenypingcap @c4pt0r PTAL again

File: tikv/get-started/install-tikv-using-binary.md [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+    ./bin/tikv-server --pd-endpoints="127.0.0.1:2379" \
+                    --addr="127.0.0.1:20162" \
+                    --data-dir=tikv3 \
+                    --log-file=tikv3.log
+    ```
+
+You can use the [pd-ctl](https://github.com/pingcap/pd/tree/master/pdctl) tool to verify whether PD and TiKV are successfully deployed:
+
+```
+./bin/pd-ctl store -d -u http://127.0.0.1:2379
```
Comment (siddontang) [inline tikv/get-started/install-tikv-using-binary.md]: seem we need to say that if we find that all the state of TiKV is "up", we can think the deployment is successful.

Comment (lilin90) [reply tikv/get-started/install-tikv-using-binary.md]: Added. Thanks!

**Comment by lilin90** [comment]
@c4pt0r PTAL again. Thanks!

**Comment by queenypingcap** [comment]
LGTM

**Pull request merged**